### PR TITLE
Redux setup

### PR DESCRIPTION
**Pull request by jogren**
#### What's this PR do?
Redux setup. Also added lock color functionality, and the ability to POST new projects and get them to display on the DOM.

#### Where should the reviewer start?
Review the Redux flow from actions to reducers. Start in app - additional changes were made in ProjectsContainer and currentColors.

#### How should this be manually tested?
Run npm start. Generate colors and create new projects.

### CODE REVIEW

File: src/util/apiCalls.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+export const getAllProjects = async () => {
+  const url = 'https://palette-picker-api-sfjo.herokuapp.com/api/v1/projects';
+  const response = await fetch(url);
+  if (!response.ok) {
+    throw Error('There was an issue getting your projects');
+  }
+
+  const projects = await response.json();
+  console.log(projects);
```
Comment (SamanthaLFreeman) [inline src/util/apiCalls.js]: Please remove the console.log

File: src/CurrentColors/CurrentColors.js [hunk 2]
```diff
@@ -1,18 +1,24 @@
...
 import Color from '../Color/Color';
+import { connect } from 'react-redux';
 
-const CurrentColors = ({colors}) => {
-  let currentColors = colors.map(color => {
-    const style = {
-      backgroundColor: color
+const CurrentColors = ({ currentPalette, toggleLock }) => {
+  let currentColors = currentPalette.map(color => {
+    const colorStyle = {
```
Comment (SamanthaLFreeman) [inline src/CurrentColors/CurrentColors.js]: Think we could make this inline

**Review by SamanthaLFreeman — CHANGES_REQUESTED** [review]
Hey Jacob,

Added two small notes for you to clean up.

File: src/CurrentColors/CurrentColors.js (hunk 2 above)
Comment (jogren) [reply src/CurrentColors/CurrentColors.js]: Good call! Will do.

File: src/util/apiCalls.js (hunk 1 above)
Comment (jogren) [reply src/util/apiCalls.js]: Just removed that console.log!

**Pull request merged**